From 16f27aef429e025713d67e42713c479deb1523ad Mon Sep 17 00:00:00 2001
From: Felipe Borges <felipeborges@gnome.org>
Date: Tue, 22 Sep 2020 16:19:01 +0200
Subject: [PATCH] welcome-tutorial: Port to libhandy-1

This regresses on the indicator dots.

Since we want to be consistent with gnome-tour, I will address this
in issue #590.

Fixes: #593
---
 build-aux/flatpak/org.gnome.Boxes.json | 3 +--
 data/ui/welcome-tutorial.ui            | 3 +--
 src/main.vala                          | 2 +-
 src/meson.build                        | 2 +-
 src/welcome-tutorial.vala              | 2 +-
 subprojects/libhandy                   | 2 +-
 subprojects/libhandy.wrap              | 2 +-
 7 files changed, 7 insertions(+), 9 deletions(-)

diff --git a/data/ui/welcome-tutorial.ui b/data/ui/welcome-tutorial.ui
index d47a1e8b..d8ef1e71 100644
--- a/data/ui/welcome-tutorial.ui
+++ b/data/ui/welcome-tutorial.ui
@@ -23,10 +23,9 @@
             <property name="expand">True</property>
 
             <child>
-              <object class="HdyPaginator" id="paginator">
+              <object class="HdyCarousel" id="paginator">
                 <property name="visible">True</property>
                 <property name="animation-duration">400</property>
-                <property name="indicator-style">dots</property>
                 <property name="margin-bottom">12</property>
                 <signal name="notify::position" handler="on_position_changed"/>
 
diff --git a/src/main.vala b/src/main.vala
index 5182ad4d..bd784fe9 100644
--- a/src/main.vala
+++ b/src/main.vala
@@ -79,7 +79,7 @@ public int main (string[] args) {
         error (err.message);
     }
 
-    Hdy.init (ref args);
+    Hdy.init ();
 
     var app = new Boxes.App ();
 
diff --git a/src/meson.build b/src/meson.build
index 19e55e20..751ae2d1 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -147,7 +147,7 @@ dependencies = [
   dependency ('tracker-sparql-3.0'),
   dependency ('vte-2.91', version: '>= 0.40.2'),
   dependency ('webkit2gtk-4.0', version: '>= 2.26.0'),
-  dependency ('libhandy-0.0', version: '>= 0.0.11'),
+  dependency ('libhandy-1', version: '>= 1.0.0'),
   cc.find_library('m', required : false),
   valac.find_library ('gio-2.0-workaround', dirs: vapi_dir),
   valac.find_library ('linux'),
diff --git a/src/welcome-tutorial.vala b/src/welcome-tutorial.vala
index 1cf1c7d0..2e9f9a12 100644
--- a/src/welcome-tutorial.vala
+++ b/src/welcome-tutorial.vala
@@ -5,7 +5,7 @@
 [GtkTemplate (ui = "/org/gnome/Boxes/ui/welcome-tutorial.ui")]
 private class Boxes.WelcomeTutorial : Gtk.Dialog {
     [GtkChild]
-    private Paginator paginator;
+    private Carousel paginator;
     [GtkChild]
     private Button go_back_button;
     [GtkChild]
-- 
GitLab

